test/e2e/service.go

```diff
@@ -226,6 +226,68 @@ var _ = framework.KubeDescribe("Services", func() {
 		validateEndpointsOrFail(c, ns, serviceName, PortsByPodName{})
 	})
 
+	It("should preserve source IP while using services", func() {
+		// TODO: verify source IP preservation for LoadBalancer type services when applicable
```

Comment:
I believe you will need a separate test case for this. So put it outside of this test case.